python/pypto/ir/pass_manager.py

```
@@ -281,10 +300,35 @@ def after_pass(_pass_obj: passes.Pass, program: core_ir.Program) -> None:
+            pass_index += 1
+
+        timing_instrument = passes.CallbackInstrument(
+            before_pass=before_pass, after_pass=after_pass, name="PipelineProfiling"
+        )
+        ctx = passes.PassContext.current()
+        outer_instruments = list(ctx.get_instruments()) if ctx else []
+        level = ctx.get_verification_level() if ctx else passes.get_default_verification_level()
+        wlevel = ctx.get_warning_level() if ctx else passes.get_default_warning_level()
+        disabled = ctx.get_disabled_warnings() if ctx else passes.WarningCheckSet()
+
+        with passes.PassContext(outer_instruments + [timing_instrument], level, wlevel, disabled):
```

Ruff (0.15.9)

[warning] Consider `[*outer_instruments, timing_instrument]` instead of concatenation

Replace with `[*outer_instruments, timing_instrument]`

(RUF005)